### DISCUSSION TITLE
Alexandr ideas!

### DISCUSSION THREAD

**Discussion by mmRoshani** [mmRoshani/sequelize-typescript-migration#9] [Ideas]
- Tests 
- Security checks for dependencies
- Nestjs Module

As told me in your email these are your ideas that are very welcome!
I think maybe by making nestjs module first the community might be stronger as I can not work on this particular project till next month we can discuss the above ideas and ways to apply them!

Let's limit all our discussion here so there is a kind of documentation for other nice users/developers of this repository.

Best regards.

@alexandr2110pro

**Comment by alexandr2110pro** [comment]
Hey @mmRoshani 
I've discovered another issue. By default it creates db column names in the same casing as fields in your model class. And that is usually camelCase. While the default behaviour for the sequelize-typescript is snake_case and it is the case in the project where I want to start using it.

So I think the next PR would be for adding an option to support snake_case fields.
Next, we want to have tests and vulnerability checks to make sure we can rely on this library in production. 

And only then - the NestJS module. 

Sorry but that is the requirements I have so I'll start doing it in this order. I have to.

**Comment by alexandr2110pro** [comment]
Another thing I can think of is a pure cli version (without the file with `SequelizeTypescriptMigration.makeMigration`).
It should allow generating all the config files with various options, etc. And then another set of commands to create/update migrations. Maybe a watch mode as well. 

IMO, we need a design doc (RFC) for the CLI interface and features to get a common vision before we start implementing it.

**Comment by alexandr2110pro** [comment]
Somewhere in-between, I want to help you refactor the source code so that it is easier to read and test.

**Comment by alexandr2110pro** [comment]
Also, add prettier and more strict eslint.

**Comment by alexandr2110pro** [comment]
Let's maybe also open a project or issues so that we can move some ideas to actionable items.

**Comment by mmRoshani** [comment]
Hi,

- That's alright I agree with the test part, as I mentioned before I can not work on this project till next month. so see it as your own project 
  and do the enhancement with any order you want to be satisfied with.

- You saw the source code and may find out there is more than one handwrite! the prettier and more strict linting would be very helpful as you suggested, about two months before @anthonylzq was working on that maybe his commits can be useful

- About the `CLI` I am interested too, think about how easy it could be to generate and roll back migrations by CLI instead of running the project or handling it in DBMS manually. I will research about that and send me any likely projects you found interesting

- I get the issue part of your last commit but by opening a new project, do you mean to make a sub-project that will help us in this project?

**Reply by alexandr2110pro** [reply]
@mmRoshani, thanks for your notes. Yes, there's something to research, indeed.

By "Let's maybe also open a project or issues so that we can move some ideas to actionable items."
I mean GitHub projects (with boards) and issues (list of issues) see https://github.com/features/issues
